skfp/descriptors/constitutional.py

```diff
@@ -0,0 +1,252 @@
+from typing import Optional, Union
+
+from rdkit.Chem import AddHs, Mol
+from rdkit.Chem.Descriptors import MolWt
+from rdkit.Chem.rdMolDescriptors import CalcNumRings, CalcNumRotatableBonds
+
+
+def validate_molecule(func):
+    """
+    Decorator to validate that the molecule has at least one atom.
+
+    Raises
+    ------
+    ValueError
+        If the molecule has no atoms.
+    """
+
+    def wrapper(mol: Mol, *args, **kwargs):
+        if mol.GetNumAtoms() == 0:
+            raise ValueError(
+                f"The molecule has no atoms, {func.__name__} cannot be calculated."
+            )
+        return func(mol, *args, **kwargs)
+
+    return wrapper
+
+
+@validate_molecule
+def average_molecular_weight(mol: Mol) -> float:
+    """
+    Average Molecular Weight.
+
+    Calculates the average molecular weight of the molecule, defined as the molecular
+    weight divided by the number of atoms.
+
+    This is different from "average molecular weight" in the context of isotopes [1]_.
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecule for which the average molecular weight is to be calculated.
+
+    References
+    ----------
+    .. [1] `
+        <https://chemistry.stackexchange.com/questions/150993/discrepancy-when-calulating-mol-weights-with-chemsketch-and-python-rdkit/150999#150999>`_
+
+    Examples
+    --------
+    >>> from rdkit.Chem import MolFromSmiles
+    >>> from skfp.descriptors.constitutional import average_molecular_weight
+    >>> mol = MolFromSmiles("C1=CC=CC=C1")  # Benzene
+    >>> average_molecular_weight(mol)
+    13.019
+    """
+    return MolWt(mol) / mol.GetNumAtoms()
+
+
+@validate_molecule
+def bond_type_count(mol: Mol, bond_type: Optional[str] = None) -> int:
+    """
+    Bond Type Count.
+
+    Counts the total number of bonds of a specific type in the molecule.
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecule for which the bond count is to be calculated.
+
+    bond_type : str, optional
+        The type of bond to count. Valid options are:
+        - "SINGLE"
+        - "DOUBLE"
+        - "TRIPLE"
+        - "AROMATIC"
```

Comment:
Simpler, link to RDKit, e.g. `Valid options are RDKit bond types [2]_`, and you can link RDKit docs: https://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html#rdkit.Chem.rdchem.BondType